cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-rococo/src/tests/snowbridge.rs

```diff
@@ -0,0 +1,526 @@
+	Rococo::execute_with(|| {
+		assert_ok!(<Rococo as RococoPallet>::XcmPallet::send(
+			sudo_origin.clone(),
+			bx!(destination.clone()),
+			bx!(create_agent_xcm),
+		));
+
+		assert_ok!(<Rococo as RococoPallet>::XcmPallet::send(
+			sudo_origin,
+			bx!(destination),
+			bx!(create_channel_xcm),
+		));
+
+		type RuntimeEvent = <Rococo as Chain>::RuntimeEvent;
+
+		assert_expected_events!(
+			Rococo,
+			vec![
+				RuntimeEvent::XcmPallet(pallet_xcm::Event::Sent { .. }) => {},
+			]
+		);
+	});
+
+	BridgeHubRococo::execute_with(|| {
+		type RuntimeEvent = <BridgeHubRococo as Chain>::RuntimeEvent;
+
+		assert_expected_events!(
+			BridgeHubRococo,
+			vec![
+				RuntimeEvent::EthereumSystem(snowbridge_system::Event::CreateChannel {
+					..
+				}) => {},
+			]
+		);
+	});
+}
+
+#[test]
+fn register_token() {
+	BridgeHubRococo::fund_accounts(vec![(
+		BridgeHubRococo::sovereign_account_id_of(MultiLocation {
+			parents: 1,
+			interior: X1(Parachain(AssetHubRococo::para_id().into())),
+		}),
+		INITIAL_FUND,
+	)]);
+
+	let message_id_: H256 = [1; 32].into();
+
+	BridgeHubRococo::execute_with(|| {
+		type RuntimeEvent = <BridgeHubRococo as Chain>::RuntimeEvent;
+		type EthereumInboundQueue =
+			<BridgeHubRococo as BridgeHubRococoPallet>::EthereumInboundQueue;
+		let message = VersionedMessage::V1(MessageV1 {
+			chain_id: CHAIN_ID,
+			command: Command::RegisterToken { token: WETH.into(), fee: XCM_FEE },
+		});
+		let (xcm, _) = EthereumInboundQueue::do_convert(message_id_, message).unwrap();
+		let _ = EthereumInboundQueue::send_xcm(xcm, AssetHubRococo::para_id().into()).unwrap();
+
+		assert_expected_events!(
+			BridgeHubRococo,
+			vec![
+				RuntimeEvent::XcmpQueue(cumulus_pallet_xcmp_queue::Event::XcmpMessageSent { .. }) => {},
+			]
+		);
+	});
+
+	AssetHubRococo::execute_with(|| {
+		type RuntimeEvent = <AssetHubRococo as Chain>::RuntimeEvent;
+
+		assert_expected_events!(
+			AssetHubRococo,
+			vec![
+				RuntimeEvent::ForeignAssets(pallet_assets::Event::Created { .. }) => {},
+			]
+		);
+	});
+}
+
+#[test]
+fn send_token_to_penpal() {
+	let asset_hub_sovereign = BridgeHubRococo::sovereign_account_id_of(MultiLocation {
+		parents: 1,
+		interior: X1(Parachain(AssetHubRococo::para_id().into())),
+	});
+	BridgeHubRococo::fund_accounts(vec![(asset_hub_sovereign.clone(), INITIAL_FUND)]);
+
+	PenpalA::fund_accounts(vec![
+		(PenpalAReceiver::get(), INITIAL_FUND),
+		(PenpalASender::get(), INITIAL_FUND),
+	]);
+
+	let weth_asset_location: MultiLocation =
+		(Parent, Parent, EthereumNetwork::get(), AccountKey20 { network: None, key: WETH }).into();
+	let weth_asset_id = weth_asset_location.into();
+
+	let origin_location =
+		MultiLocation { parents: 2, interior: weth_asset_location.interior.split_last().0 };
+
+	// Fund ethereum sovereign in asset hub
+	let ethereum_sovereign: AccountId =
+		hex!("ce796ae65569a670d0c1cc1ac12515a3ce21b5fbf729d63d7b289baad070139d").into();
```

Comment:
@claravanstaden  I see you have this also as a constant in `src/inbound/tests.rs`, but I would prefer to have algo-generated `ethereum_sovereign` account, I guess you use something like this:
```
const NETWORK: NetworkId = Ethereum { chain_id: 11155111 };
let contract_location = MultiLocation { parents: 2, interior: X1(GlobalConsensus(NETWORK)) };
let account = GlobalConsensusEthereumConvertsFor::<[u8; 32]>::convert_location(&contract_location).unwrap();
```
It would be much better than hard-coded constant without description.

Anyway, in `src/inbound/tests.rs` you use `SS58Format=2` which is not the case for AHR where it is [`42`](https://github.com/paritytech/polkadot-sdk/blob/master/cumulus/parachains/runtimes/assets/asset-hub-rococo/src/lib.rs#L161), prefix does not matter for accountId bytes.
But let's not use hard-coded bytes, but better let's be clear and maybe to save someone's time in the future when investigating what those bytes represent. Also, maybe the converter could change and generate different accountId and so on.